### ISSUE TITLE
[Bug]: Error with set_global_handler("arize_phoenix")

### ISSUE THREAD

**Issue by austinmw** [run-llama/llama_index#10602]
### Bug Description

The `query_pipeline.ipynb` example shows the following:

```python
# setup Arize Phoenix for logging/observability
import phoenix as px

px.launch_app()
import llama_index

llama_index.set_global_handler("arize_phoenix")
```

This results in:

> AttributeError: module 'llama_index' has no attribute 'set_global_handler'

I tried modifying it to:

```python
# setup Arize Phoenix for logging/observability
import phoenix as px

px.launch_app()
from llama_index.core import set_global_handler

set_global_handler("arize_phoenix")
```

However I get:

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
File [/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:8](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:8), in arize_phoenix_callback_handler(**kwargs)
      [7](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:7) try:
----> [8](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:8)     from phoenix.trace.llama_index import OpenInferenceTraceCallbackHandler
      [9](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:9) except ImportError:

File [/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/__init__.py:1](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/__init__.py:1)
----> [1](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/__init__.py:1) from .callback import OpenInferenceTraceCallbackHandler
      [2](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/__init__.py:2) from .debug_callback import DebugCallbackHandler

File [/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:8](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:8)
      [4](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:4) from typing import (
      [5](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:5)     Any,
      [6](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:6) )
----> [8](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:8) from openinference.instrumentation.llama_index._callback import (
      [9](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:9)     OpenInferenceTraceCallbackHandler as _OpenInferenceTraceCallbackHandler,
     [10](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:10) )
     [11](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:11) from openinference.instrumentation.llama_index.version import (
     [12](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:12)     __version__,
     [13](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/phoenix/trace/llama_index/callback.py:13) )

File [/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/__init__.py:5](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/__init__.py:5)
      [3](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/__init__.py:3) from typing import TYPE_CHECKING, Any, Collection
----> [5](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/__init__.py:5) from openinference.instrumentation.llama_index._callback import OpenInferenceTraceCallbackHandler
      [6](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/__init__.py:6) from openinference.instrumentation.llama_index.package import _instruments

File [/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:40](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:40)
     [38](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:38) from wrapt import ObjectProxy
---> [40](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:40) from llama_index.callbacks.base_handler import BaseCallbackHandler
     [41](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:41) from llama_index.callbacks.schema import (
     [42](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:42)     BASE_TRACE_EVENT,
     [43](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:43)     CBEventType,
     [44](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:44)     EventPayload,
     [45](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/openinference/instrumentation/llama_index/_callback.py:45) )

ModuleNotFoundError: No module named 'llama_index.callbacks.base_handler'

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
Cell In[3], [line 7](vscode-notebook-cell:?execution_count=3&line=7)
      [4](vscode-notebook-cell:?execution_count=3&line=4) px.launch_app()
      [5](vscode-notebook-cell:?execution_count=3&line=5) from llama_index.core import set_global_handler
----> [7](vscode-notebook-cell:?execution_count=3&line=7) set_global_handler("arize_phoenix")

File [/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:11](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:11), in set_global_handler(eval_mode, **eval_params)
      [8](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:8) """Set global eval handlers."""
      [9](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:9) import llama_index.core
---> [11](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:11) llama_index.core.global_handler = create_global_handler(eval_mode, **eval_params)

File [/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:51](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:51), in create_global_handler(eval_mode, **eval_params)
     [45](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:45)     except ImportError:
     [46](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:46)         raise ImportError(
     [47](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:47)             "ArizePhoenixCallbackHandler is not installed. "
     [48](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:48)             "Please install it using `pip install llama-index-callbacks-arize-phoenix`"
     [49](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:49)         )
---> [51](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:51)     handler = arize_phoenix_callback_handler(**eval_params)
     [52](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:52) elif eval_mode == "honeyhive":
     [53](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/core/callbacks/global_handlers.py:53)     try:

File [/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:10](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:10), in arize_phoenix_callback_handler(**kwargs)
      [8](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:8)     from phoenix.trace.llama_index import OpenInferenceTraceCallbackHandler
      [9](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:9) except ImportError:
---> [10](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:10)     raise ImportError(
     [11](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:11)         "Please install Arize Phoenix with `pip install -q arize-phoenix`"
     [12](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:12)     )
     [13](https://vscode-remote+ssh-002dremote-002bdev.vscode-resource.vscode-cdn.net/opt/conda/lib/python3.10/site-packages/llama_index/callbacks/arize_phoenix/base.py:13) return OpenInferenceTraceCallbackHandler(**kwargs)

ImportError: Please install Arize Phoenix with `pip install -q arize-phoenix`
```

Even though this library is already installed.

### Version

0.10.1

### Steps to Reproduce

Added steps in description

**Comment by nerdai** [comment]
Hi @austinmw are you using a fresh virtualenv? With v0.10.xx we're now using Python namespaces, whereas v0.9.xx and earlier did not. So any remnants of legacy llama-index in the virtualenv will break things.

So essentially, the recommendation is to:

1. use a fresh virtualenv
2. install either `llama-index` (or `llama-index-core`)
3. install the other necessary llama-index packages for your script/app

**Comment by logan-markewich** [comment]
In this case, for arize, you'll also want to install `pip install llama-index-callbacks-arize-phoenix`

**Comment by austinmw** [comment]
@logan-markewich Hey, I actually did install that as well. I think the install error is a false issue.

**Comment by logan-markewich** [comment]
@austinmw ahhh I see now.

I think arize is just incompatible with v0.10.0 at the moment (just got off a call with some of their team, they are working on it)

**Comment by logan-markewich** [comment]
If anyone is using LlamaIndex 0.9.xx but still want to use phoenix they can, they just have to downgrade the instrumentation `openinference-instrumentation-llama-index<1.0.0`

**Comment by babinux** [comment]
Any chance you could update the Colab that was shared in that YT video? https://www.youtube.com/watch?v=CeDS1yvw9E4&t=211s

**Comment by thanhtheman** [comment]
after installing the above packages, the probem is not fixed with llama-index 0.10.34
